xwiki-platform-core/xwiki-platform-localization/xwiki-platform-localization-rest/xwiki-platform-localization-rest-default/src/test/java/org/xwiki/localization/internal/DefaultTranslationsResourceTest.java

```diff
@@ -0,0 +1,238 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.localization.internal;
+
+import java.io.StringWriter;
+import java.util.Locale;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mock;
+import org.xwiki.localization.LocalizationContext;
+import org.xwiki.localization.LocalizationManager;
+import org.xwiki.localization.Translation;
+import org.xwiki.localization.rest.model.jaxb.ObjectFactory;
+import org.xwiki.localization.rest.model.jaxb.Translations;
+import org.xwiki.model.ModelContext;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.test.LogLevel;
+import org.xwiki.test.junit5.LogCaptureExtension;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+
+import ch.qos.logback.classic.Level;
+
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test of {@link DefaultTranslationsResource}.
+ *
+ * @version $Id$
+ * @since 13.3RC1
+ */
+@ComponentTest
+class DefaultTranslationsResourceTest
+{
+    private static final DocumentReference CURRENT_ENTITY =
+        new DocumentReference("currentwiki", "XWiki", "CurrentEntity");
+
+    @InjectMockComponents
+    private DefaultTranslationsResource defaultTranslationResource;
+
+    @MockComponent
+    private LocalizationContext localizationContext;
+
+    @MockComponent
+    private LocalizationManager localizationManager;
+
+    @MockComponent
+    private ModelContext modelContext;
+
+    private final Locale defaultLocale = Locale.forLanguageTag("es");
+
+    @RegisterExtension
+    LogCaptureExtension logCapture = new LogCaptureExtension(LogLevel.WARN);
+
+    @Mock
+    private Translation translationKey1;
+
+    @Mock
+    private Translation translationKey2;
+
+    @BeforeEach
+    void setUp()
+    {
+        when(this.localizationContext.getCurrentLocale()).thenReturn(this.defaultLocale);
+        when(this.modelContext.getCurrentEntityReference()).thenReturn(CURRENT_ENTITY);
+    }
+
+    @Test
+    void getTranslations() throws Exception
+    {
+        ObjectFactory objectFactory = new ObjectFactory();
+        Translations expected = objectFactory.createTranslations();
```

Comment:
These are used in many of the tests below so I would move them as fields in the test class.

Reply:
I'm not sure it's simplifying the tests, especially regarding readability and changeability.

Reply:
Let's keep it like that then.